custom_components/ha-google-home/api.py

```diff
@@ -52,62 +54,88 @@ def get_android_id(self):
         """Fetches timers and alarms from google device"""
-        url = self.create_url(device.ip, device.port, endpoint)
-        _LOGGER.debug(
-            "For device {device} - {url}".format(device=device.device_name, url=url)
-        )
-        # verify=False is need to avoid SSL security checks. Otherwise it will fail to connect"""
-        HEADERS[HEADER_CAST_LOCAL_AUTH] = device.local_auth_token
-        response = requests.get(url, headers=HEADERS, verify=False, timeout=TIMEOUT)
-
-        if response.status_code != HTTP_OK:
-            _LOGGER.error(
-                "For device {device} - API returned {error}: {reason}".format(
-                    device=device.device_name,
-                    error=response.status_code,
-                    reason=response.text,
-                )
+        url = self.create_url(device.ip, PORT, endpoint)
+        if url:
+            _LOGGER.debug(
+                "For device {device} - {url}".format(device=device.device_name, url=url)
             )
-            return
+            # verify=False is need to avoid SSL security checks. Otherwise it will fail to connect"""
+            HEADERS[HEADER_CAST_LOCAL_AUTH] = device.local_auth_token
+            response = requests.get(url, headers=HEADERS, verify=False, timeout=TIMEOUT)
+
+            if response.status_code != HTTP_OK:
+                _LOGGER.error(
+                    "For device {device} - API returned {error}: {reason}".format(
+                        device=device.device_name,
+                        error=response.status_code,
+                        reason=response.text,
+                    )
+                )
+                return
+            else:
+                return response.json()
         else:
-            return response.json()
+            return
 
     def get_google_devices_information(self):
         """Retrieves devices from glocaltokens"""
         _LOGGER.debug("Fetching data...")
+        offline_devices = []
         devices = self._client.get_google_devices()
 
         for device in devices:
             # To avoid key error's
             timers = []
             alarms = []
-
-            result = self.get_alarms_and_timers_from(device, API_ENDPOINT_ALARMS)
-            if result:
-                timers = result.get(TIMERS)
-                alarms = result.get(ALARMS)
-
-                if not timers and not alarms:
-                    _LOGGER.error(
-                        "For device {device} - {error}".format(
-                            device=device.device_name, error=API_RETURNED_UNKNOWN
+            if device.ip:
+                result = self.get_alarms_and_timers_from(device, API_ENDPOINT_ALARMS)
+                if result:
+                    timers = result.get(TIMERS)
+                    alarms = result.get(ALARMS)
+
+                    if not timers and not alarms:
+                        _LOGGER.error(
+                            "For device {device} - {error}".format(
```

Comment:
Shall we make the error message more elaborate "Failed to fetch timers/alarms information from device {device} ..."?

Reply:
It should not be anything that causes concern to the user, this will happen regularly because the request sometimes fails for no good reason. But it could probably be worded better then what i have done. But i think that goes for all error, debug and warning messages so far :smile:

Reply:
Actually, shouldn't we use exceptions to make error messages?

Reply:
It didn't seem from the documentation that HA wants custom integrations to raise exceptions, so the logger is a good way to go about it :)